portfolio/src/main/java/com/google/sps/servlets/PieChartServlet.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet stores and retrieves data about favorite milk tea flavors. */
+@WebServlet("/pie-chart")
+public class PieChartServlet extends HttpServlet {
+    private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String flavor = request.getParameter("flavor");
+        String topping = request.getParameter("topping");
+
+        Entity voteEntity = new Entity("BobaVote");
```

Comment:
if this were production code i would ask why you are storing individual votes instead of immediately aggregating them.  there will generally be far more requests to view results than there will be votes, so it makes sense to optimize for reading instead of writing to the data store here. i don't want you to get bogged down in the details however, and this code is otherwise perfectly fine!

Reply:
I changed the TeaVoteServlet to optimize for reading instead of writing as suggested. Datastore now stores the votes for each flavor and topping rather than each individual vote.